src/api/gtfsService.ts

```diff
@@ -118,9 +130,41 @@ export async function getGtfsStopHitTimesAsync(stop: BusStop, timestamp: Moment)
     .slice(0, MAX_HITS_COUNT)
 
   const rideIds = closestInTimeRides.map((ride) => ride.id).join(JOIN_SEPARATOR)
-  const stopHits = await GTFS_API.gtfsRideStopsListGet({
-    gtfsRideIds: rideIds,
-    gtfsStopIds: stop.stopId.toString(),
-  })
-  return stopHits.sort((hit1, hit2) => +hit1.arrivalTime! - +hit2.arrivalTime!)
+
+  const minStartTime = Math.min(...rides.map((ride) => Number(ride.startTime)))
+  const maxEndTime = Math.max(...rides.map((ride) => Number(ride.endTime)))
+
+  /* Fix StopHits bugs next steps TODO:
+  1. Add a test to ensure this feature is working correctly.
+  2. Optimize the axios request to minimize latency. - Currently takes forever.
+  3. Fix any on this- define the hit type
+  */
+
+  try {
+    const stopHitsRequestPayLoad: StopHitsPayLoadType = {
+      gtfsRideIds: rideIds,
+      gtfsStopIds: stop.stopId.toString(),
+      arrival_time_from: minStartTime,
+      arrival_time_to: maxEndTime,
+    }
+
+    const stopHitsRes = await axios.get(`${BASE_PATH}/gtfs_ride_stops/list`, {
+      params: stopHitsRequestPayLoad,
+    })
+
+    if (stopHitsRes.status !== 200) {
+      throw new Error(`Error fetching stop hits: ${stopHitsRes.statusText}`)
+    }
+
+    if (stopHitsRes.data.length === 0) {
+      throw new Error(`No stop hits found`)
+    }
+
+    const stopHits: GtfsRideStopPydanticModel[] = stopHitsRes.data
+
+    return stopHits.sort((hit1, hit2) => +hit1.arrivalTime! - +hit2.arrivalTime!)
+  } catch (error) {
+    console.error(`Error fetching stop hits:`, error)
```

Comment:
nice. I guess we can use some error boundary component to render a meaning full error message to the user